### ISSUE TITLE
Deleting the first function deletes the source table

### ISSUE THREAD

**Issue by nairabhishek73** [DalgoT4D/webapp#632]
https://jam.dev/c/053693b8-1549-494b-b32f-dd9864611de4

Source table should be retained when the function is deleted

**Comment by fatchat** [comment]
@Ishankoradia this is where you said we have to create nodes in the database right
but sources already have `OrgDbtModel`s associated to them
and there is no edge to add
so can this not be done in the frontend? not across sessions, just when they delete the first op coming out of a source node

**Comment by Ishankoradia** [comment]
No, this cant be done on the frontend. When we delete the first operation in the chain connected to the source, we also delete the target model since no operations are now available. This also deletes the edge from source to this operation node. 

Our graph with the current logic only renders the nodes that have edges, standalone nodes are not returned.

**Comment by fatchat** [comment]
i was thinking could we maintain in the frontend a list of sources which have been added to the canvas, and after refreshing the graph manually re-insert the ones which aren't in the graph

not for the first release though

**Comment by Ishankoradia** [comment]
you re-insert mean all of them ?

**Comment by fatchat** [comment]
Yes, the ones which aren’t in the graph response

Of course they might be reinserted in different places so we would have to
save their positions, and make sure to put them onto the canvas before the
nodes from the backend

Bit of an effort

**Comment by Ishankoradia** [comment]
Hmm i see.

Creating dummy nodes in the backend sounds more simpler and they can also delete from the graph if they dont want to see if again & again on every refresh. Have graph return these nodes.
